analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -74,12 +66,11 @@ public ExecutingSqlQueries()
                 : null;
 
         protected override bool IsConcat(ExpressionSyntax argument, SemanticModel semanticModel) =>
-            IsStringMethodInvocation("Concat", argument, semanticModel) ||
-            (
-                argument.IsKind(SyntaxKind.AddExpression) &&
-                argument is BinaryExpressionSyntax concatenation &&
-                !IsConcatenationOfConstants(concatenation, semanticModel)
-            );
+            IsStringMethodInvocation("Concat", argument, semanticModel)
+            || IsConcatenation(argument, semanticModel)
+            || (argument is IdentifierNameSyntax identifierNameSyntax
+                && semanticModel.GetDeclaringSyntaxNode(identifierNameSyntax) is VariableDeclaratorSyntax variableDeclaratorSyntax
```

Comment:
This condition doesn't seem right. We're inside `IsConcat(ExpressionSyntax)` context. And the condition checks if the parent is variable declarator with initializer and recursion. That feels hacky.

I'd expect some `AddExpression` syntax to be checked here.

Reply:
This looks the same way as it did before. Am I missing something?

Reply:
Let's refactor it. It's a nightmare to maintain.

The Base `IsConcat`, `IsFormat` and `IsInterpolated` should be merged to single `IsTracked`, while removing `IsStringMethodInvocation` indirection. That will also remove 6 of those private helpers.

The ObjectCreation and PropertyAccess can be then embedded directly into the tracker, as it's not reused (just rename the `context =>` to `c =>` for example)

On the C#/VB side, the variable declarator step interpolation can be decoupled and simplified as well.